### ISSUE TITLE
proxy: router at capacity error

### ISSUE THREAD

**Issue by jamesallen-vol** [linkerd/linkerd2#2363]
### What is the issue?
I'm seeing logging errors stating the proxy router is at capacity from the linkerd-controller

### How can it be reproduced?
I'm currently running 9 meshed services. 8/9 deployments are running 2-3 pods. 1/9 of the deployments is running approximately 150 pods.

Checking the metrics of the linkerd controller using `kubectl exec linkerd-controller -n linkerd -c linkerd-proxy curl localhost:4191/metrics | wc -l
` produces `6754`

### Logs, error output, etc
"ERR! proxy={server=out listen=127.0.0.1:4140 remote=10.69.3.253:42516} linkerd2_proxy::proxy::http::router router at capacity (100)"

These errors are produced approximately every 30 minutes.

#### `linkerd check` output

```text
kubernetes-api
--------------
√ can initialize the client
√ can query the Kubernetes API

kubernetes-version
------------------
√ is running the minimum Kubernetes API version

linkerd-existence
-----------------
√ control plane namespace exists
√ controller pod is running
√ can initialize the client
√ can query the control plane API

linkerd-api
-----------
√ control plane pods are ready
√ can query the control plane API
√ [kubernetes] control plane can talk to Kubernetes
√ [prometheus] control plane can talk to Prometheus

linkerd-service-profile
-----------------------
√ no invalid service profiles

linkerd-version
---------------
√ can determine the latest version
√ cli is up-to-date

control-plane-version
---------------------
√ control plane is up-to-date
√ control plane and cli versions match

Status check results are √

```

### Environment

- Kubernetes Version:  v1.10.11-gke.1
- Cluster Environment:  GKE
- Host OS:  Container-Optimized OS 
- Linkerd version: 2.2.1

**Comment by olix0r** [comment]
Thanks for the report, @jamesallen-vol 

Can you describe a bit about the HTTP requests that this pod sends? How many services is this app a client of? How does the app set the `Host` header (if HTTP/1) for these requests?

Additionally, if you have a pod in this state, I'd be curious to see the output of curling `$pod:4191/metrics` (or running `kubectl exec ...pod... curl localhost:4191/metrics`)

**Comment by kleimkuhler** [comment]
@jamesallen-vol I’m trying to figure out if this behavior should be expected with default configurations, or if you are experiencing a bug in the router logic. If you still have the proxy router in a state that you would be to answer some of Oliver’s questions above, that would be super helpful!

Also, is the application is doing any of its own service discovery or load balancing? Are you aware of any scenario where the would be requests directly to an IP instead of the service that is trying to be reached?

**Comment by klingerf** [comment]
It looks like this issue was fixed in linkerd/linkerd2-proxy#203, and we'll bump the proxy dependency in this repo prior to this week's edge release to pick it up.

**Comment by hawkw** [comment]
Should be fixed in `edge-19.3.1`!
